### PR TITLE
orchestrate: SetPod in reality after launching

### PR DESCRIPTION
**Pull request by petertseng**

### CODE REVIEW

**Comment by petertseng** [comment]
too simple?!

**Comment by anthonybishopric** [comment]
lgtm

**Comment by tummychow** [comment]
Code lgtm but the corresponding GetPod also needs to be implemented (it will probably replace the pod.CurrentManifest on line 154). Might as well do that in this PR

**Comment by tummychow** [comment]
Discovered bug while deploying to universe cluster. In `reality.Pod`, a 404 will cause a nil pointer dereference. (this bug was actually introduced in #54 but the code path wasn't being exercised until now)

I originally guessed that the consul-api package would propagate a 404 back to the caller if the key didn't exist. Unfortunately I was [wrong](https://github.com/armon/consul-api/blob/master/kv.go#L42). If the key doesn't exist, then the returned KVPair and error are *both* nil. Then we try to dereference the nil pointer on line 60 (`kvPair.Value`) and that causes an explosion.

Correct behavior would be to detect `kvPair == nil` and return a nil pod manifest.

File: pkg/reality/reality.go [hunk 1]
```diff
@@ -57,6 +57,9 @@ func (r *Store) Pod(node, podId string) (*pods.PodManifest, error) {
 	if err != nil {
 		return nil, err
 	}
+	if kvPair == nil {
+		return nil, pods.NoCurrentManifest
```
Comment (anthonybishopric) [inline pkg/reality/reality.go]: :+1:

**Comment by tummychow** [comment]
lgtm, also tested on universe and verified

**Comment by anthonybishopric** [comment]
lg

**Pull request merged**